tests/performance/event_generator/tests/test_logeventgenerator.py

```diff
@@ -0,0 +1,187 @@
+# Copyright (C) 2024, Wazuh Inc.
+# Created by Wazuh, Inc. <info@wazuh.com>.
+# This program is a free software; you can redistribute it and/or modify it under the terms of GPLv2
+
+"""Unit tests for the LogEventGenerator class."""
+
+import json
+import os
+from pathlib import Path
+
+import pytest
+
+from event_generator import LogEventGenerator
+
+
+@pytest.fixture
+def log_generator(tmp_path: Path) -> LogEventGenerator:
+    """Fixture that provides a LogEventGenerator instance with a unique log file for each test.
+
+    Args:
+        tmp_path (LocalPath): Temporary directory path fixture provided by pytest.
+    """
+    path = tmp_path / "test.log"
+    generator = LogEventGenerator(
+        rate=1,
+        path=str(path),
+        operations=5,
+        max_file_size=1,
+        template_path=None
+    )
+    return generator
+
+
+def test_log_file_exists(log_generator: LogEventGenerator) -> None:
+    """Test if the log file is created."""
+    generator = log_generator
+    generator.start()
+    assert os.path.exists(generator.path), "Log file should exist after generation."
+
+
+def test_log_event_count(log_generator: LogEventGenerator) -> None:
+    """Test if the correct number of log events are generated."""
+    generator = log_generator
+    generator.start()
+    with open(generator.path) as file:
+        lines = file.readlines()
+    assert len(lines) == generator.operations, f"Exactly {generator.operations} log entries should be written."
+
+
+def test_log_content(log_generator: LogEventGenerator) -> None:
+    """Test the content of the log file."""
+    generator = log_generator
+    generator.start()
+    with open(generator.path) as file:
+        content = file.read()
+    assert "This is a test log message" in content, "Log message should be in the log file."
+
+
+@pytest.mark.parametrize("rate, operations", [
+    (1, 5),
+    (10, 50),
+    (100, 1000),
+])
+def test_log_event_count_parametrized(tmp_path: Path, rate: int, operations: int) -> None:
+    """Test if the correct number of log events are generated with different rates and operations.
+
+    Args:
+        tmp_path (LocalPath): Temporary directory path fixture provided by pytest.
+        rate (int): Number of operation per second.
+        operations (int): number of times logs will be generated.
+    """
+    path = tmp_path / "test.log"
+    generator = LogEventGenerator(
+        rate=rate,
+        path=str(path),
+        operations=operations,
+        max_file_size=None,
+        template_path=None
+    )
+    generator.start()
+    with open(path) as file:
+        lines = file.readlines()
+    assert len(lines) == operations, f"Exactly {operations} log entries should be written."
```

Comment:
The rate option is not tested here. We cannot guarantee that events are being generated in the provided files at the specified rate. Please check the amount of time taken by the generator to produce these events and ensure it matches the expected rate

Reply:
Fixed here: https://github.com/wazuh/wazuh-qa/pull/5697/commits/6c6cccdd6742801840e4fc95582c082f1ab73f21